pkg/proxy/ipvs/netlink_linux.go

```diff
@@ -169,15 +154,24 @@ func (h *netlinkHandle) GetLocalAddresses(dev, filterDev string) (sets.String, e
 	filterMask := netlink.RT_FILTER_TABLE | netlink.RT_FILTER_TYPE | netlink.RT_FILTER_PROTOCOL
 
 	// find chosen device
-	if chosenLinkIndex != -1 {
+	if chosenLinkIndex, err := h.getLinkIndex(dev); err != nil {
+		return nil, err
+	} else if -1 != chosenLinkIndex {
```

Comment:
I don't think this is the typical pattern in kubernetes, `chosenLinkIndex != -1` is preferred than this.

Reply:
i'll change it later

Reply:
This else if is not needed because the `if` above returns on err

Reply:
@andrewsykim hi，only when `chosenLinkIndex != -1`,  it can be assigned to `routeFilter.LinkIndex`
and  `chosenLinkIndex == -1` is not a kind of error here，like its defination here ：
```go
func (h *netlinkHandle) getLinkIndex(dev string) (int, error) {
	if "" == dev {
		return -1, nil
	}
```
i think that just mean excuting ` ip r show table ......`  without followed ` dev xxx`

How do you think of this？

Reply:
I see what you mean, but wouldn't this still acheive the same thing? :
```go
if chosenLinkIndex, err := getLinkIndex(h, dev); err != nil {
		return nil, err
} 

if chosenLinkIndex != -1 {
		routeFilter.LinkIndex = chosenLinkIndex
		routeFilter.LinkIndex = chosenLinkIndex
		filterMask |= netlink.RT_FILTER_OIF
		filterMask |= netlink.RT_FILTER_OIF
}
```